### PR TITLE
Removing the struct/unmanaged constraint from the Vector types

### LINKED ISSUE #85893: [API Proposal]: Remove struct constraint from `Vector*<T>`
The constraint makes it much more challenging to use.  For example, you can't access `Vector*<T>.IsSupported`, you can't write routines that first check whether T is a known / supported type and then use `VectorX<T>.Create(T value)`, etc.

According to @tannergooding, this constraint used to help improve codegen but with improvements already made to the JIT, we should be able to remove it without negative impact on code quality.

### PR DESCRIPTION
**Pull request by tannergooding**
This resolves #85893

This does not actually utilize the relaxed constraint anywhere yet. Give the churn involved in removing the constraints, I opted to leave that to a future PR.

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Numerics/Vector.cs [hunk 1]
```diff
@@ -1055,8 +1015,8 @@ public static Vector<T> LoadAligned<T>(T* source)
         [Intrinsic]
         [CLSCompliant(false)]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static Vector<T> LoadAlignedNonTemporal<T>(T* source)
-            where T : unmanaged => LoadAligned(source);
+        public static Vector<T> LoadAlignedNonTemporal<T>(T* source) => LoadAligned(source);
+#pragma warning restore CS8500 // This takes the address of, gets the size of, or declares a pointer to a managed type ('T')
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Numerics/Vector.cs:1019]: This is unrelated, yes?  It's fine, just confirming.

Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/src/System/Numerics/Vector.cs:1019]: Oh, nevermind, it is related, it was relying on unmanaged being a subset of struct.

File: src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector64.cs [hunk 2]
```diff
@@ -2373,9 +2309,8 @@ public static Vector64<T> Subtract<T>(Vector64<T> left, Vector64<T> right)
         [Intrinsic]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static T Sum<T>(Vector64<T> vector)
-            where T : struct
         {
-            T sum = default;
+            T sum = default!;
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector64.cs:2313]: This makes me realize this change could actually introduce new warnings in 3rd party code.  That's fine, just something to keep in mind.  I'd been thinking of the removal of the constraint as being completely non-breaking.

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector64.cs (hunk 2 above)
Comment (tannergooding) [reply src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector64.cs:2313]: I don't think that's the case....

This warning is because we made an explicit change to the constraint such that its now unconstrained and now could be `string`, therefore `default` would be invalid for the specified `T`.

Existing callers must be `where T : struct`, so the compiler would consider `T` to be always non-nullable and the result to likewise be always non-nullable.

Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector64.cs:2313]: > Existing callers must be where T : struct

Ah, of course.

**Comment by SamMonoRT** [comment]
cc @lambdageek

**Comment by vargaz** [comment]
I don't think this will affect mono.
